### PR TITLE
fix: use electron-osx-sign instead of manual code signing

### PR DESCRIPTION
**Pull request by joaomoreno**

### CODE REVIEW

File: build/gulpfile.vscode.darwin.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+const path = require('path');
+
+gulp.task('vscode-codesign', async () => {
+	const buildDir = process.env['AGENT_BUILDDIRECTORY'];
+
+	if (!buildDir) {
+		throw new Error('$AGENT_BUILDDIRECTORY not set');
+	}
+
+	if (process.env.VSCODE_QUALITY !== 'insider' && process.env.VSCODE_QUALITY !== 'stable') {
```
Comment (joaomoreno) [inline build/gulpfile.vscode.darwin.js:20]: We also have `exploration`, and potentially more in the future, so I'd like not to lock us down here. Could we simply avoid this check and derivate the app name from the fs?

File: build/gulpfile.vscode.darwin.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+const gulp = require('gulp');
+const { signAsync } = require('electron-osx-sign');
+const util = require('./lib/util');
+const path = require('path');
+
+gulp.task('vscode-codesign', async () => {
```
Comment (joaomoreno) [inline build/gulpfile.vscode.darwin.js:13]: At this point, there's no reason for this to actually be a gulp task, we're not using any task dependencies nor streaming functionalities. Let's simply move all of this into a `build/darwin/sign.ts` file (typescript would be great) and we'd simply invoke it with `node build/darwin/sign.js`.

File: build/azure-pipelines/darwin/product-build-darwin.yml [hunk 3]
```diff
@@ -93,64 +93,64 @@ steps:
...
-  condition: and(succeeded(), eq(variables['VSCODE_STEP_ON_IT'], 'false'))
-
-- script: |
-    set -e
-    yarn test-browser --build --browser chromium --browser webkit --browser firefox
-  displayName: Run unit tests (Browser)
-  condition: and(succeeded(), eq(variables['VSCODE_STEP_ON_IT'], 'false'))
-
-- script: |
+#- script: |
```
Comment (joaomoreno) [inline build/azure-pipelines/darwin/product-build-darwin.yml:96]: I know these comments are only temporary, but you can always change the `VSCODE_STEP_ON_IT` to anything but `false` when submitting a build, to skip all of these steps.

File: build/gulpfile.vscode.darwin.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+	const defaultOpts = {
+		app: path.join(appRoot, appName),
+		platform: 'darwin',
+		entitlements: path.join(__dirname, 'azure-pipelines', 'darwin', 'app-entitlements.plist'),
+		hardenedRuntime: true,
+		'pre-auto-entitlements': false,
+		'pre-embed-provisioning-profile': false,
+		keychain: path.join(buildDir, 'buildagent.keychain'),
+		version: util.getElectronVersion(),
+		identity: '99FM488X57',
```
Comment (joaomoreno) [inline build/gulpfile.vscode.darwin.js:41]: Yeah I believe this is correct, we'd have to see how this translates to the actual `codesign` cli command, to compare to what was happening before.

Comment (joaomoreno) [reply build/gulpfile.vscode.darwin.js:41]: Is maybe the new thing calling codesign under a different user?

Comment (joaomoreno) [reply build/gulpfile.vscode.darwin.js:41]: There seems to be a [`debuglog` utility](https://github.com/electron/electron-osx-sign/blob/63d7b60cfae6e2c830f7e2bda8d376d547544c85/sign.js#L385), let's enable it and see if we learn more: https://github.com/electron/electron-osx-sign#debug

Comment (deepak1556) [reply build/gulpfile.vscode.darwin.js]: https://monacotools.visualstudio.com/Monaco/_build/results?buildId=75004

**Comment by deepak1556** [comment]
The logging was helpful,

**Error** - `No identity found for signing.`

**Command Used** - `security find-identity -v /Users/runner/runners/2.166.4/work/1/buildagent.keychain
`


But the actual keychain path is at https://dev.azure.com/monacotools/Monaco/_build/results?buildId=75009&view=logs&j=a5e52b91-c83f-5429-4a68-c246fc63a4f7&t=1d952d4e-e27f-5005-50bf-de7d93606599&l=13

I thought `path.join(buildDir, 'buildagent.keychain')` would be this, but that doesn't seem to the case.

**Comment by joaomoreno** [comment]
Try `path.join(process.env['AGENT_TEMPDIRECTORY'], 'buildagent.keychain')`

**Comment by deepak1556** [comment]
@joaomoreno signing issues are fixed now, but mooncake step is failing.

What is this step for ?

Why `yarn` would fail inside `build/` when the parent compile step was fine ?

**Comment by joaomoreno** [comment]
Mooncake only installs deps in `build/package.json`. Since `electron-osx-sign` is really build specific let's move that dependency over there.

**Comment by deepak1556** [comment]
Thanks for the context!

New build https://monacotools.visualstudio.com/DefaultCollection/Monaco/_build/results?buildId=75286&view=results

**Comment by deepak1556** [comment]
Smoke tested with `yarn smoketest --build --stable-build`.

@joaomoreno this is ready for review. Thanks!

**Review by joaomoreno — COMMENTED** [review]
Changes look good. Smoke tested the build and all seems to run fine. 👍  Good job

**Comment by deepak1556** [comment]
Thanks!

**Pull request merged**